ksql-parser/src/test/java/io/confluent/ksql/parser/rewrite/StatementRewriterTest.java

```diff
@@ -0,0 +1,616 @@
+    Statement statement = KSQL_PARSER.buildAst(queryStr, metaStore).get(0);
+
+    StatementRewriter statementRewriter = new StatementRewriter();
+    Statement rewrittenStatement = (Statement) statementRewriter.process(statement, null);
+
+    assertThat("testSelectTumblingWindow failed.", rewrittenStatement, instanceOf(Query.class));
+    Query query = (Query) rewrittenStatement;
+    assertThat("testSelectTumblingWindow failed.", query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
+    assertThat("testCreateTable failed.", querySpecification.getSelect().getSelectItems
+        ().size() == 2);
+    assertThat("testSelectTumblingWindow failed.", querySpecification.getWhere().get().toString(), equalTo("(ORDERS.ORDERUNITS > 5)"));
+    assertThat("testSelectTumblingWindow failed.", ((AliasedRelation)querySpecification.getFrom()).getAlias(), equalTo("ORDERS"));
+    assertThat("testSelectTumblingWindow failed.", querySpecification
+        .getWindowExpression().isPresent());
+    assertThat("testSelectTumblingWindow failed.", querySpecification
+        .getWindowExpression().get().toString(), equalTo(" WINDOW STREAMWINDOW  TUMBLING ( SIZE 30 SECONDS ) "));
+  }
+
+  @Test
+  public void testSelectHoppingWindow() {
+
+    String
+        queryStr =
+        "select itemid, sum(orderunits) from orders window HOPPING ( size 30 second, advance by 5"
+            + " seconds) "
+            + "where "
+            + "orderunits"
+            + " > 5 group by itemid;";
+    Statement statement = KSQL_PARSER.buildAst(queryStr, metaStore).get(0);
+
+    StatementRewriter statementRewriter = new StatementRewriter();
+    Statement rewrittenStatement = (Statement) statementRewriter.process(statement, null);
+
+    assertThat(rewrittenStatement, instanceOf(Query.class));
+    Query query = (Query) rewrittenStatement;
+    assertThat(query.getQueryBody(), instanceOf(QuerySpecification.class));
+    QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
+    assertThat(querySpecification.getSelect().getSelectItems().size(), equalTo(2));
+    assertThat(querySpecification.getWhere().get().toString(), equalTo("(ORDERS.ORDERUNITS > 5)"));
+    assertThat(((AliasedRelation)querySpecification.getFrom()).getAlias().toUpperCase(), equalTo("ORDERS"));
+    assertThat("window expression isn't present", querySpecification
+        .getWindowExpression().isPresent());
+    assertThat(querySpecification.getWindowExpression().get().toString().toUpperCase(),
+        equalTo(" WINDOW STREAMWINDOW  HOPPING ( SIZE 30 SECONDS , ADVANCE BY 5 SECONDS ) "));
+  }
+
+  @Test
+  public void should() {
+    List<Statement> statements = KSQL_PARSER.buildAst("select * from orders;", metaStore);
+    System.out.println(statements);
+  }
+  @Test
```

Comment:
nit: spacing between functions

Reply:
That test was removed since it was not doing anything.